src/arch/linux/mapping/mapping.c

```diff
@@ -280,8 +280,13 @@ int alias_mapping(int cap, dosaddr_t targ, size_t mapsize, int protect, void *so
     target = MEM_BASE32x(targ, i);
     if (target == MAP_FAILED)
       continue;
-    /* protections on KVM_BASE go via page tables in the VM, not mprotect */
-    prot = i == KVM_BASE ? (PROT_READ|PROT_WRITE|PROT_EXEC) : protect;
+    prot = protect;
+    if (i == KVM_BASE) {
+      /* in KVM targ is physical memory, not equal to virtual memory for high mem */
+      if (targ >= ALIAS_SIZE) continue;
```

Comment:
This almost suggests you need both
`pa_size` and `va_size` fields in mem_bases.
Not insisting though.

Reply:
The only place where this is relevant for alias_mapping() I could find is DPMI mappings to conventional memory (int31/ax=0509). In that case this would be better done by adjusting KVM page tables, so it's already a bit of a special case that should be adjusted (but it's a bit beyond the scope of this PR).